lib/rainforest/cli/test_importer.rb

```diff
@@ -224,15 +186,117 @@ def validate
   def create_new file_name = nil
     name = @options.file_name if @options.file_name
     name = file_name if !file_name.nil?
+    ext = test_files.file_extension
 
     uuid = SecureRandom.uuid
-    name = "#{uuid}#{EXT}" unless name
-    name += EXT unless name[-EXT.length..-1] == EXT
-    name = File.join([@options.test_spec_folder, name])
+    name = "#{uuid}#{ext}" unless name
+    name += ext unless name[-ext.length..-1] == ext
+    name = File.join([@test_files.test_folder, name])
 
     File.open(name, "w") { |file| file.write(sprintf(SAMPLE_FILE, uuid)) }
 
     logger.info "Created #{name}" if file_name.nil?
     name
   end
+
+  private
+
+  def upload_groups_sequentially(upload_groups)
+    progress_bar = ProgressBar.create(title: 'Rows', total: test_files.count, format: '%a %B %p%% %t')
+    upload_groups.each_with_index do |rfml_tests, idx|
+      if idx == (rfml_tests.length - 1)
+        upload_group_in_parallel(rfml_tests, progress_bar)
+      else
+        rfml_tests.each { |rfml_test| upload_test(rfml_test) }
+        progress_bar.increment
+      end
+    end
+  end
+
+  def upload_group_in_parallel(rfml_tests, progress_bar = nil)
+    progress_bar ||= ProgressBar.create(title: 'Rows', total: rfml_tests.count, format: '%a %B %p%% %t')
+    Parallel.each(rfml_tests, in_threads: THREADS, finish: lambda { |item, i, result| progress_bar.increment }) do |rfml_test|
+      upload_test(rfml_test)
+    end
+  end
+
+  def upload_test(rfml_test)
+    return unless rfml_test.steps.count > 0
+
+    if @options.debug
+      logger.debug "Starting: #{rfml_test.rfml_id}"
+      logger.debug "\t#{rfml_test.start_uri || "/"}"
+    end
+
+    test_obj = create_test_obj(rfml_test)
+    # Upload the test
+    begin
+      if rfml_id_mappings[rfml_test.rfml_id]
+        t = Rainforest::Test.update(rfml_id_mappings[rfml_test.rfml_id], test_obj)
+
+        logger.info "\tUpdated #{rfml_test.rfml_id} -- ##{t.id}" if @options.debug
+      else
+        t = Rainforest::Test.create(test_obj)
+
+        logger.info "\tCreated #{rfml_test.rfml_id} -- ##{t.id}" if @options.debug
+        rfml_id_mappings[rfml_test.rfml_id] = t.id
+      end
+    rescue => e
+      logger.fatal "Error: #{rfml_test.rfml_id}: #{e}"
+      exit 2
+    end
+  end
+
+  def rfml_id_mappings
+    if @_id_mappings.nil?
+      @_id_mappings = {}.tap do |id_mappings|
+        Rainforest::Test.all(page_size: 1000, rfml_ids: test_files.rfml_ids).each do |rf_test|
+          rfml_id = rf_test.rfml_id
+          next if rfml_id.nil?
+
+          id_mappings[rfml_id] = rf_test.id
+        end
+      end
+    end
+    @_id_mappings
+  end
+
+  def create_test_obj(rfml_test)
+    test_obj = {
+      start_uri: rfml_test.start_uri || "/",
+      title: rfml_test.title,
+      description: rfml_test.description,
+      tags: (["ro"] + rfml_test.tags).uniq,
```

Comment:
Not yet. I'm just trying to get this working for now for some clients who are waiting on this to create their tests. That'll mostly be a server/frontend change I think.